test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -4461,6 +4461,34 @@ public async Task CustomIMessageSerializerSettingsFactory()
             }
         }
 
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task CustomSerializerSettings_TypeNameHandlingAll()
+        {
+            string[] orchestratorFunctionNames =
+            {
+                nameof(TestOrchestrations.SayHelloInline),

```

Comment:
This test passes even with the customer serializer settings before your source code changes. In order to make sure this code tests the bug fix, we need to run an orchestrator that uses a more complicated type.

Reply:
Oh, oops, I forgot to use an orchestrator that uses an activity function. That is where the input gets turned into an object[] in DTFx and where/why this serialization issue occurs.

Reply:
Good catch!